### PR TITLE
Load multiple packages in pyodide.loadPackage

### PR DESCRIPTION
**Pull request by rth**
This solves part of https://github.com/iodide-project/pyodide/issues/135 and allows loading multiple packages in a single `pyodide.loadPackage` call,
```py
pyodide.loadPackage(['numpy', 'matplotlib'])
```
I'm am a bit bothered by the singular of "package". Maybe ideally it could have been another method `pyodide.loadPackages` or more neutral `pyodide.load`? Generally, in the use case where we have a list of dependencies and want to load them, this could be useful. The analogy with `pip install` that can install one or multiple packages also works.  WDYT ?

This PR also adds a unit test `test_load_packages_simultaneous` (that fails on master) corresponding to the case where we make two `pyodide.loadPackage` calls leading to an error. I'm  not quite sure how to address that (and my understanding of concurrent programming in JS is fairly limited). One possibility could have been to store a reference to the promise, created in `loadPackage` and if the reference is not `null`, wait for it to complete in any subsequent `loadPackage` call. The issue is that there is quite a bit of code in `loadPackage` that happens before the promise is returned (even assuming that this approach would work).

### CODE REVIEW

File: test/test_package_loading.py [hunk 1]
```diff
@@ -34,3 +35,30 @@ def test_invalid_package_name(selenium):
...
+def test_load_packages_simultaneous(selenium_standalone):
+    selenium = selenium_standalone
+
+    from selenium.common.exceptions import TimeoutException
+
+    selenium.run_js(
+        'window.done = false\n'
+        'pyodide.loadPackage("numpy")\n'
+        'pyodide.loadPackage("matplotlib")'
+        '.then(function() { window.done = true; })')
```
Comment (mdboom) [inline test/test_package_loading.py]: This sort of thing *could* get flaky on Chrome. On Chrome, we are forced to load the wasm asynchronously, but unfortunately that uses global variables to set up, so you can't actually load wasm modules simultaneously.  All that happens in emscripten itself, but I don't think it actually does any locking around simultaneous loading.

We may want to add our own locking around this case.  i.e., if there's already an active promise, attach the new promise to run *after* that promise.

File: test/test_package_loading.py [hunk 2]
```diff
@@ -34,3 +35,30 @@ def test_invalid_package_name(selenium):
     with pytest.raises(WebDriverException,
                        match="Invalid package name or URI"):
         selenium.load_package('tcp://some_url')
+
+
+def test_load_packages_multiple(selenium_standalone):
+    selenium = selenium_standalone
+    selenium.load_package(['pyparsing', 'matplotlib'])
```
Comment (mdboom) [inline test/test_package_loading.py]: Should we test something that aren't dependencies of one another, like `matplotlib` and `py`?

File: test/test_package_loading.py (hunk 1 above)
Comment (rth) [reply test/test_package_loading.py]: Thanks for the review and your comments !  I responded in https://github.com/iodide-project/pyodide/issues/135#issuecomment-418402486 and removed this failing test: as I haven't found a solution for this yet, I would rather make a subsequent PR for it...

File: test/test_package_loading.py (hunk 2 above)
Comment (rth) [reply test/test_package_loading.py]: That was intentional to check that in the log output we only see `Loading pyparsing` once (it wasn't the case before changes in this PR). Pushed another commit to also test non dependent packages as you suggest.

**Comment by rth** [comment]
I think I addressed all comments, and also added some documentation.

**Comment by mdboom** [comment]
This looks good.  I'm going to keep #135 open as a reminder that calling `loadPackage` multiple times sequentially can cause issues.  I think we can/should fix that bug as well, though this approach is handy as well.

**Pull request merged**